homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -250,6 +250,24 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
         await self.async_set_unique_id(unique_id)
         self._abort_if_unique_id_configured(updates={CONF_URL: url})
 
+        def _is_supported_device() -> bool:
+            """
+            See if we are looking at a possibly supported device.
+
+            Matching solely on SSDP data does not yield reliable enough results.
+            """
+            try:
+                with Connection(url=url, timeout=CONNECTION_TIMEOUT) as conn:
+                    basic_info = Client(conn).device.basic_information()
```

Comment:
Should this also be executed inside an executor, as I assume it performs blocking I/O?

Reply:
The `_is_supported_device` function where this is in, has its only call site on line 268 below, and that runs the whole function in an executor.

Reply:
Oh, indeed, sorry for the noise.